ament_flake8/ament_flake8/main.py

```diff
@@ -0,0 +1,224 @@
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    return rc
+
+
+def generate_flake8_report(config_file, paths, excludes, max_line_length=None):
+    kwargs = {
+        'repeat': True,
+        'show_source': True,
+        'verbose': True,
+        'reporter': CustomReport,
+        'config_file': config_file,
+    }
+    if max_line_length is not None:
+        kwargs['max_line_length'] = max_line_length
+
+    # add options for flake8 plugins
+    kwargs['parser'], options_hooks = flake8.engine.get_parser()
+    flake8style = CustomStyleGuide(**kwargs)
+    options = flake8style.options
+    for options_hook in options_hooks:
+        options_hook(options)
+
+    if excludes:
+        flake8style.options.exclude += excludes
+
+    # flake8 uses a wrapper StyleGuide to handle particular OSErrors
+    kwargs['styleguide'] = flake8style
+    wrapperStyleGuide = flake8.engine.StyleGuide(**kwargs)
+
+    return wrapperStyleGuide.check_files(paths)
+
+
+def get_xunit_content(report, testname):
+    data = {
+        'testname': testname,
+        'test_count': max(report.total_errors, 1),
+        'error_count': report.total_errors,
+        'time': '%.3f' % round(report.elapsed, 3),
+    }
+    xml = '''<?xml version="1.0" encoding="UTF-8"?>
+<testsuite
+  name="%(testname)s"
+  tests="%(test_count)d"
+  failures="%(error_count)d"
+  time="%(time)s"
+>
+''' % data
+
+    if report.errors:
+        # report each flake8 error/warning as a failing testcase
+        for error in report.errors:
+            data = {
+                'quoted_name': quoteattr(
+                    error['error_code'] +
+                    ' (%(path)s:%(row)d:%(column)d)' % error),
+                'testname': testname,
+                'quoted_message': quoteattr(
+                    '%(error_message)s:\n%(source_line)s' % error),
+            }
+            xml += '''  <testcase
+    name=%(quoted_name)s
+    classname="%(testname)s"
+  >
+      <failure message=%(quoted_message)s/>
+  </testcase>
+''' % data
+
+    else:
+        # if there are no flake8 errors/warnings report a single successful test
+        data = {
+            'testname': testname,
+        }
+        xml += '''  <testcase
+    name="flake8"
+    classname="%(testname)s"
+    status="No errors or warnings"/>
+''' % data
+
+    # output list of checked files
+    data = {
+        'escaped_files': escape(''.join(['\n* %s' % f for f in report.files])),
+    }
+    xml += '''  <system-out>Checked files:%(escaped_files)s</system-out>
+''' % data
+
+    xml += '</testsuite>\n'
+    return xml
+
+
+class CustomStyleGuide(flake8.engine.NoQAStyleGuide):
+
+    def input_file(self, filename, **kwargs):
+        self.options.reporter.files.append(filename)
+        return super(CustomStyleGuide, self).input_file(filename, **kwargs)
+
+
+class CustomReport(pep8.StandardReport):
```

Comment:
Is there a way to implement this without using `pep8` directly (which the package doesn't declare a dependency on)?